spec/lib/api/redfish-1.0/system-spec.js

```diff
@@ -1294,47 +1329,183 @@ describe('Redfish Systems Root', function () {
+                .expect(500)
+                .expect(function() {
+                    expect(redfish.handleError.called).to.be.true;
+                });
+        });
 
-    it('should 404 an invalid processor', function() {
-        return helper.request().get('/redfish/v1/Systems/' + node.id + '/Processors/bad')
-            .expect('Content-Type', /^application\/json/)
-            .expect(404);
-    });
+        it('should return a valid processor for a DELL node', function() {
+            waterline.catalogs.findLatestCatalogOfSource.resolves(Promise.resolve({
+                node: 'Dell1234abcd1234abcd1234abcd',
+                source: 'dummysource',
+                data: dellCatalogData
+            }));
+            redfish.getVendorNameById.resolves(Promise.resolve({vendor: 'Dell'}));
+
+            return helper.request().get('/redfish/v1/Systems/' + dellNode.id + '/Processors/0')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function() {
+                    expect(tv4.validate.called).to.be.true;
+                    expect(validator.validate.called).to.be.true;
+                    expect(redfish.render.called).to.be.true;
+                });
+        });
 
+        it('should throw an invalid socketId for a DELL node', function() {
+            waterline.catalogs.findLatestCatalogOfSource.resolves(Promise.resolve({
+                node: '1234abcd1234abcd1234abcd',
+                source: 'dummysource',
+                data: dellCatalogData
+            }));
+            redfish.getVendorNameById.resolves(Promise.resolve({vendor: 'Dell'}));
+
+            return helper.request().get('/redfish/v1/Systems/' + dellNode.id + '/Processors/100')
+                .expect('Content-Type', /^application\/json/)
+                .expect(404)
+                .expect(function(err) {
+                    expect(err.body.error['@Message.ExtendedInfo'][1].Message).to.be.equal('invalid socketId');
+                });
+        });
+
+        it('should return a valid processor for a Cisco node', function() {
+            waterline.catalogs.findLatestCatalogOfSource.resolves(Promise.resolve({
+                node: '1234abcd1234abcd1234abcd',
+                source: 'dummysource',
+                data: ucsCatalog
+            }));
+            waterline.nodes.needByIdentifier.withArgs(ucsNode.id).resolves(Promise.resolve(ucsNode));
+            redfish.getVendorNameById.resolves(Promise.resolve({vendor: 'Cisco'}));
+
+            return helper.request().get('/redfish/v1/Systems/' + ucsNode.id + '/Processors/0')
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .expect(function() {
+                    expect(tv4.validate.called).to.be.true;
+                    expect(validator.validate.called).to.be.true;
+                    expect(redfish.render.called).to.be.true;
```

Comment:
Is it possible to check a specific field from response body to see whether it could return correct info to users based on the mocked catalog data?